Test/Unit/Model/ReclaimTest.php

```diff
@@ -0,0 +1,266 @@
+            $loggerMock,
+            $scopeSettingMock,
+            self::TEST_LOG_PATH
+        );
+
+        /**
+         * Create new Reclaim with mocked arguments
+         */
+        $this->object = new Reclaim(
+            $objectManagerMock,
+            $quoteFactoryMock,
+            $productFactoryMock,
+            $stockItemMock,
+            $stockItemRepositoryMock,
+            $subscriberMock,
+            $subscriberCollectionMock,
+            $scopeSettingMock,
+            $loggerHelperMock
+        );
+
+        /**
+         * create test log file with dummy entries
+         */
+        $testLogFile = fopen(self::TEST_LOG_PATH, 'wb');
+        chmod(self::TEST_LOG_PATH, 0644);
+        foreach (self::TEST_ENTRIES as $entry)
+        {
+            fwrite($testLogFile, $entry . "\r\n");
+        }
+        fclose($testLogFile);
+    }
+
+    protected function tearDown()
+    {
+        unlink(self::TEST_LOG_PATH);
+    }
+
+    public function testReclaimInstance()
+    {
+        $this->assertInstanceOf(Reclaim::class, $this->object);
+    }
+
+    public function testReclaim()
+    {
+        $this->assertSame('1.1.9', $this->object->reclaim());
+    }
+
+    public function testGetLog()
+    {
+        /**
+         * test successful retrieval
+         */
+        $testLog = file(self::TEST_LOG_PATH);
+        $this->assertSame($testLog, $this->object->getLog());
+
+        /**
+         * test unsuccessful retrieval scenarios
+         */
+        $expectedResponse = array (
+            'message' => 'Unable to retrieve log file with error: file(' . self::TEST_LOG_PATH . '): failed to open stream: No such file or directory'
+        );
+        unlink(self::TEST_LOG_PATH);
+        $this->assertSame($expectedResponse, $this->object->getLog());
+
+        $testLogFile = fopen(self::TEST_LOG_PATH, 'wb');
+        chmod(self::TEST_LOG_PATH, 0644);
+        fclose($testLogFile);
+        $expectedResponse = array (
+            'message' => 'Log file is empty'
+        );
+        $this->assertSame($expectedResponse, $this->object->getLog());
+    }
+
+    public function testCleanLog()
+    {
+        /**
+         * Test when invalid date is provided
+         */
+        $badDateString = 'asdf123asdfa';
+        $expectedResponse = array(
+            'message' => 'Unable to parse timestamp: ' . $badDateString
+        );
+        $this->assertSame($expectedResponse, $this->object->cleanLog($badDateString));
+
+        /**
+         * Test when valid date is provided
+         */
+        $validDateString = '2020-01-01 00:00:00';
+        $expectedResponse = $response = array(
+            'message' => 'Cleaned all log entries before: ' . $validDateString
+        );
+        $this->assertSame($expectedResponse, $this->object->cleanLog($validDateString));
+
+        //checking side effects
+        $testLog = file(self::TEST_LOG_PATH);
+        $this->assertSame($testLog, $this->object->getLog());
+    }
```

Comment:
At least for python i think each of these assertions should be their own tests e.g. testCleanLogInvalidDatInput, testCleanLogValidDateInput. No big deal if there's a convention otherwise for php.